calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -22,36 +24,33 @@
 from calico_ctl.pool import validate_arguments as pool_validate_arguments
 from calico_ctl.profile import validate_arguments as profile_validate_arguments
 from calico_ctl.container import validate_arguments as container_validate_arguments
+from calico_ctl.utils import validate_cidr, validate_ip, validate_characters
 from pycalico.datastore_datatypes import BGPPeer
 
 
 class TestBgp(unittest.TestCase):
 
-    @patch('sys.exit', autospec=True)
-    def test_validate_arguments(self, m_sys_exit):
+    @parameterized.expand([
+        ({'<PEER_IP>':'127.a.0.1'}, False),
+        ({'<PEER_IP>':'aa:bb::zz'}, False),
+        ({'<AS_NUM>':9}, True),
+        ({'<AS_NUM>':'9'}, True),
+        ({'<AS_NUM>':'nine'}, False),
+        ({'show':1, '--ipv4':1}, True)
+    ])
+    def test_validate_arguments(self, case, expected_result):
         """
         Test validate_arguments for calicoctl bgp command
         """
-        # Set up arguments
-        cases = (
-            ({'<PEER_IP>':'127.a.0.1'}, False),
-            ({'<PEER_IP>':'aa:bb::zz'}, False),
-            ({'<AS_NUM>':9}, True),
-            ({'<AS_NUM>':'9'}, True),
-            ({'<AS_NUM>':'nine'}, False),
-            ({'show':1, '--ipv4':1}, True)
-        )
-
-        # Call method under test for each test case
-        # Assert that method exits on bad input
-        for case, is_valid in cases:
-            print "Testing case %s ..." % case
+        with patch('sys.exit', autospec=True) as m_sys_exit:
+            # Call method under test
             bgp_validate_arguments(case)
-            if is_valid:
-                assert not m_sys_exit.called
+
+            # Assert that method exits on bad input
```

Comment:
expected_result is an odd name since bgp_validate_arguments doesn't return a result.

Perhaps if it was sys_exit_called, you could simplify the following as:

self.assertEquals(m_sys_exit.called, sys_exit_called)